src/Common/src/Interop/Unix/libcrypto/Interop.ERR.cs

```diff
@@ -17,13 +19,56 @@ internal static partial class libcrypto
         [DllImport(Libraries.LibCrypto, CharSet = CharSet.Ansi)]
         private static extern void ERR_error_string_n(uint e, StringBuilder buf, int len);
 
-        internal static string GetOpenSslErrorString()
+        private static string ERR_error_string_n(uint error)
         {
-            uint error = ERR_get_error();
             StringBuilder buf = new StringBuilder(1024);
-            
+
             ERR_error_string_n(error, buf, buf.Capacity);
             return buf.ToString();
         }
+
+        internal static Exception CreateOpenSslCryptographicException()
+        {
+            const uint ReasonMask = 0x0FFF;
+            const uint ERR_R_MALLOC_FAILURE = 64 | 1;
+
+            uint error = ERR_get_error();
+            uint last = error;
+
+            while (last != 0)
+            {
+                last = ERR_get_error();
```

Comment:
Hmm. I'm not sure where it'd make sense to log these. Inner exceptions is an interesting idea; the concern is that they may be (likely are?) unrelated to the current error.